backend_new/test/integration/user.e2e-spec.ts

```diff
@@ -101,4 +117,416 @@ describe('User Controller Tests', () => {
 
     expect(res.body.id).toEqual(userA.id);
   });
+
+  it('should update user when user exists', async () => {
+    const userA = await prisma.userAccounts.create({
+      data: await userFactory(),
+    });
+
+    const res = await request(app.getHttpServer())
+      .put(`/user/${userA.id}`)
+      .send({
+        id: userA.id,
+        firstName: 'New User First Name',
+        lastName: 'New User Last Name',
+      } as UserUpdate)
+      .expect(200);
+
+    expect(res.body.id).toEqual(userA.id);
+    expect(res.body.firstName).toEqual('New User First Name');
+    expect(res.body.lastName).toEqual('New User Last Name');
+  });
+
+  it("should error when updating user that doesn't exist", async () => {
+    await prisma.userAccounts.create({
+      data: await userFactory(),
+    });
+    const randomId = randomUUID();
+    const res = await request(app.getHttpServer())
+      .put(`/user/${randomId}`)
+      .send({
+        id: randomId,
+        firstName: 'New User First Name',
+        lastName: 'New User Last Name',
+      } as UserUpdate)
+      .expect(404);
+
+    expect(res.body.message).toEqual(
+      `user ${randomId} was requested but not found`,
+    );
+  });
+
+  it('should delete user when user exists', async () => {
+    const userA = await prisma.userAccounts.create({
+      data: await userFactory(),
+    });
+
+    const res = await request(app.getHttpServer())
+      .delete(`/user/`)
+      .send({
+        id: userA.id,
+      } as IdDTO)
+      .expect(200);
+
+    expect(res.body.success).toEqual(true);
+  });
+
+  it("should error when deleting user that doesn't exist", async () => {
+    const randomId = randomUUID();
+    const res = await request(app.getHttpServer())
+      .delete(`/user/`)
+      .send({
+        id: randomId,
+      } as IdDTO)
+      .expect(404);
+
+    expect(res.body.message).toEqual(
+      `user ${randomId} was requested but not found`,
+    );
+  });
+
+  it('should resend confirmation when user exists', async () => {
+    const userA = await prisma.userAccounts.create({
+      data: await userFactory(),
+    });
+
+    const res = await request(app.getHttpServer())
+      .post(`/user/resend-confirmation/`)
+      .send({
+        email: userA.email,
+        appUrl: 'https://www.google.com',
+      } as EmailAndAppUrl)
+      .expect(201);
+
+    expect(res.body.success).toEqual(true);
+
+    const userPostResend = await prisma.userAccounts.findUnique({
+      where: {
+        id: userA.id,
+      },
+    });
+
+    expect(userPostResend.email).toBe(userA.email);
+    expect(userPostResend.confirmationToken).not.toBeNull();
+  });
+
+  it('should succeed when trying to resend confirmation but not update record when user is already confirmed', async () => {
```

Comment:
What are the use cases where we need to resend the confirmation if the user is already confirmed?

Reply:
Hey sean we don't actually resend the the email or modify the user record at all. Instead we just return success as true and do nothing else. 

If I had to guess its probably to hide if a user is already confirmed or not